ingestion_tools/scripts/data_validation/conftest.py

```diff
@@ -35,35 +36,56 @@ def run_names(bucket: str, dataset: str, run_glob: str) -> List[str]:
     return run_names
 
 
-def voxel_spacings(bucket: str, dataset: str, voxelspacing_glob: str) -> List[float]:
-    """All voxel spacings matching the vozel spacing glob pattern in the dataset."""
+def get_voxel_spacing_files(bucket: str, dataset: str, run_names: List[str], voxelspacing_glob: str) -> List[str]:
     fs: S3Filesystem = FileSystemApi.get_fs_api(mode="s3", force_overwrite=False)
-    tentatives = fs.glob(f"s3://{bucket}/{dataset}/*/Tomograms/{voxelspacing_glob}")
+
+    tentatives = []
+
+    # Use ThreadPoolExecutor to run glob_task in parallel
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        # Map the run_names to the glob_task function
+        results = executor.map(
+            lambda run_name: fs.glob(f"s3://{bucket}/{dataset}/{run_name}/Tomograms/{voxelspacing_glob}"),
+            run_names,
+        )
+
+        # Combine results
+        for result in results:
+            tentatives += result
 
     exclude = ["Images", "dataset_metadata.json"]
 
     for ex in exclude:
         tentatives = [tent for tent in tentatives if ex not in tent]
```

Comment:
This check should be a part of the fixture that generates the run_names and not be done here. [Which it already is](https://github.com/chanzuckerberg/cryoet-data-portal-backend/blob/1deb4ab6c71c3347ee9b77c5fdcff4b412ce62c8/ingestion_tools/scripts/data_validation/conftest.py#L32-L34).

Reply:
fixed in a later pr, see #250